src/DependencyInjection/Nette/NetteContainer.php

```diff
@@ -31,13 +32,13 @@ public function getService(string $serviceName)
 		return $this->container->getService($serviceName);
 	}
 
-	/**
-	 * @param string $className
-	 * @return mixed
-	 */
 	public function getByType(string $className)
 	{
-		return $this->container->getByType($className);
+		$service = $this->container->getByType($className);
+		if ($service === null) {
+			throw new ShouldNotHappenException(sprintf('Service for class %s not found.', $className));
+		}
+		return $service;
```

Comment:
phpstan reports

> Method PHPStan\DependencyInjection\Nette\NetteContainer::getByType() should return T of object but returns object.

I dont know how to fix that

Reply:
https://phpstan.org/user-guide/stub-files

Reply:
for now [I committed a workaround](https://github.com/phpstan/phpstan-src/pull/759/commits/3ba82ab592db3ee4c01821876756f71ea743a366), which no longer requires the stubfile - since I was not able to resolve the error mentioned in the comment above.

I figured the problem regarding "should return T of object but returns T of object." should be solved in a separate PR...?

Reply:
i dislike inline `@var` when there can be a better solution: https://phpstan.org/writing-php-code/phpdocs-basics#inline-%40var

So I'd like the stub file much more. And this PR is fine, it's called "More precise phpdocs in Container".

Reply:
ok. I reverted the last 3 commits and we are back at the point 

> Method PHPStan\DependencyInjection\Nette\NetteContainer::getByType() should return T of object but returns T of object.